### PR TITLE
Sort declarations when generating Python bindings

### PR DESCRIPTION
**Pull request by apaszke**
This helps resolve ambiguities in argument parsing according to
any rules we will need.

For now, this allows us to make scalar operations more conservarive
wrt. argument types, but makes them commutative again.

@gchanan

### CODE REVIEW

**Comment by apaszke** [comment]
Looks like the change has also uncovered a bug in `remainder`:
```python
x = torch.tensor([1, 2, 3, 4, 5])
y = torch.tensor(-1)
print(x % y.expand_as(x))
print(x % y.item())
```

**Comment by apaszke** [comment]
I have a fix for it, but will also need to add a test. Will push it today in the evening

File: aten/src/THC/THCTensorMathPointwise.cuh [hunk 1]
```diff
@@ -462,22 +476,22 @@ struct TensorCRemainderOp {
 template <>
 struct TensorCRemainderOp<float> {
   __device__ __forceinline__ void operator()(float* out, float* in) {
-    *out = *in != 0 ? *out - *in * floorf(*out / *in) : NAN;
+    *out = *in != 0.f ? *out - *in * floorf(*out / *in) : NAN;
```
Comment (gchanan) [inline aten/src/THC/THCTensorMathPointwise.cuh]: shouldn't 0 be okay in these cases?  It should automatically be cast to the right type?

**Review by gchanan — APPROVED** [review]

File: aten/src/THC/THCTensorMathPointwise.cuh (hunk 1 above)
Comment (apaszke) [reply aten/src/THC/THCTensorMathPointwise.cuh]: It probably is. I only changed it to be more consistent, it's not part of the fix really.

**Pull request merged**